### ISSUE TITLE
Running 1.0.1 will be stopped in a certain height

### ISSUE THREAD

**Issue by kyleO-O** [bitcoin-sv/bitcoin-sv#148]
I met a problem.
I ran the 1.0.1 node client, after syncing to a certain height (such as 460000 or 540000), the client will be stopped. At this time, after I restart, the client will start to sync from the first block every time.
How to fix it?

**Comment by kyleO-O** [comment]
I checked bitcoind.log and there was no error for this

**Comment by Danconnolly** [comment]
Are there any errors in the log related to storing the blocks on disk?

Are you using docker?

**Comment by kyleO-O** [comment]
There was no error for this in bitcoind.log. And I ran 1.0.1 client in the host , not using docker.
After I tried, I found, the longer the client runs, the greater its memory consumption.

Does the program crash with memory leaks?

Is my 1.0.1 client program stopped because of insufficient memory? Stopping at this time will cause the synchronized block to fail to record.  So after restarting, synchronization will start from the first block.

![issue](https://user-images.githubusercontent.com/42598610/74086692-4af57d00-4ac0-11ea-8640-ee78592121aa.png)

**Comment by bitcartel** [comment]
@kyleO-O How much memory does your host/computer have?

**Comment by kyleO-O** [comment]
32G.
But before that, my host also had other programs running, which also took up some memory. And it caused the 1.0.1 client to easily crash. 
So, I shut down other programs that use memory, so that the 1.0.1 client does not crash temporarily.

**Comment by kyleO-O** [comment]
One situation is this: 
I actively shut down the client (at this time the synchronized block height is 140000). When I restart the client, the block height will continue to sync from 140,000. 
Keep watching until the client crashes and shuts down (at this time, the synchronized block height is about 560000). I restart and the client will still resynchronize from 140000, not 560000 or 1.

So I guess the program did crash and failed to record the block heights that were synchronized at the time. And the active shutdown of the program will record the block height synchronized at that time.

I hope this description will help you find the cause of these problems.

**Comment by daviunx** [comment]
> Are there any errors in the log related to storing the blocks on disk?
> 
> Are you using docker?

@Danconnolly Is there any issue related to running the node with docker? I'm having the same issue and i'm running the node using docker.

Thanks.

**Comment by peterlayke** [comment]
I am having the same issue. Just watched a node hit about 540000 (ish), stop and continue from 129400.

This is running on Ubuntu on a VM 16GB RAM.

**Comment by daviunx** [comment]
Hi guys,

I've been able to unblock the download setting to unlimited excessive block size and max stack memory usage:
```
excessiveblocksize=0
maxstackmemoryusageconsensus=0
maxstackmemoryusagepolicy=0
```

I'm using docker bitcoinsv testnet.

**Comment by scrawld** [comment]
@Danconnolly For docker version 1.0.2, it stops at runtime with no error logging, no response to RPC API requests before it stops, and no high concurrent requests

**Comment by scrawld** [comment]
This is my configuration file
```
rpcuser=test
rpcpassword=test
rpcallowip=0.0.0.0/0
rpcport=9555
rpctimeout=30
rpcworkqueue=125

server=1
#daemon=1
#reindex=1
txindex=1
testnet=0
dbcache=8000
datadir=/data
#printtoconsole=1

excessiveblocksize=2000000000
maxstackmemoryusageconsensus=200000000
```

**Comment by scrawld** [comment]
@Danconnolly 
When the node did not respond, I found this log in the bitcoind.log. I felt that the node was trying to connect, resulting in no response.
Do you have a solution? Maybe it's my problem.
```
2020-02-24 06:37:27 [bitcoin-opencon] connect() to 152.254.255.4:8333 failed after select(): Connection refused (111)
2020-02-24 06:43:44 [bitcoin-opencon] connect() to [2607:5300:203:1e25::]:8333 failed: Network is unreachable (101)
2020-02-24 06:48:11 [bitcoin-net] socket sending timeout: 1201s
2020-02-24 06:48:11 [bitcoin-net] socket sending timeout: 1201s
2020-02-24 06:51:28 [bitcoin-opencon] connect() to 69.4.92.250:8333 failed after select(): Connection refused (111)
```

**Comment by scrawld** [comment]
@Danconnolly Running to a certain height in version 1.0.2 will still stop, and there will be no error logs

**Comment by scrawld** [comment]
@Danconnolly Is there a time node for this problem

**Comment by Danconnolly** [comment]
Created new issue for @scrawld , #159 

If anyone else is still having issues syncing, then please open a new issue.
